graph/path/internal/testgraphs/shortest.go

```diff
@@ -644,7 +658,13 @@ var ShortestPathTests = []struct {
 		HasNegativeWeight: true,
 		HasNegativeCycle:  true,
 
-		Query: simple.Edge{F: simple.Node(0), T: simple.Node(3)},
+		Query:  simple.Edge{F: simple.Node(0), T: simple.Node(3)},
+		Weight: math.Inf(-1),
```

Comment:
If this is a "special" value, perhaps it should be a constant.

Reply:
It's not special and since these are test cases, things should be explicit.